spec.go

```diff
@@ -78,3 +78,13 @@ type Hook struct {
 	Args []string `json:"args"`
 	Env  []string `json:"env"`
 }
+
+// State holds information about the runtime state of the container
```

Comment:
We introduce the `Spec` entry with:

> Spec is the base configuration for the container.  It specifies platform independent configuration.

Since the `State` type is the root of a new JSON structure, independent of `Spec`, we may want to echo that here with something like:

> State is the base type for the container's runtime state.  It specifies platform independent state.

Or something like that to help folks avoid confusing the spec types and the state types.